🐛 BUG: C3 failed to deploy a project - ✘ [ERROR] Not logged in. - stale token?

### Which Cloudflare product(s) does this pertain to?

C3

### What version of `Wrangler` are you using?

create-cloudflare@2.0.0-next.2 + wrangler@0.0.0-d8f8f3d2/2.20.0

### What operating system are you using?

Mac

### Describe the Bug

I tried to run the following on a machine which I haven't used in a long time but previously used wrangler on:
```
npm create cloudflare@next
```

I created and tried to deploy a simple (all defaults) Angular and Qwik project. Both failed with `✘ [ERROR] Not logged in.` error:

<summary>
C3 + Angular
<details>
npm create cloudflare@next
Need to install the following packages:
  create-cloudflare@2.0.0-next.2
Ok to proceed? (y) 

using create-cloudflare version 2.0.0-next.1

╭ Create an application with Cloudflare Step 1 of 3
│ 
├ Where do you want to create your application?
│ dir ilovejavascript.net
│
├ What type of application do you want to create?
│ type webFramework
│
├ Which development framework do you want to use?
│ framework Angular
│
╰ Continue with Angular Creating your application via `npx @angular/cli@next new ilovejavascript.net --standalone`

Need to install the following packages:
  @angular/cli@16.0.0
Ok to proceed? (y) 
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
? Would you like to enable autocompletion? This will set up your terminal so pressing TAB while typing Angular CLI 
commands will show possible options and autocomplete arguments. (Enabling autocompletion will modify configuration files
 in your home directory.) Yes
Appended `source <(ng completion script)` to `/Users/igor/.zshrc`. Restart your terminal or run the following to autocomplete `ng` commands:

    source <(ng completion script)
? Would you like to share pseudonymous usage data about this project with the Angular Team
at Google under Google's Privacy Policy at https://policies.google.com/privacy. For more
details and how to change this setting, see https://angular.io/analytics. Yes

Thank you for sharing pseudonymous usage data. Should you change your mind, the following
command will disable this feature entirely:

    ng analytics disable --global

Global setting: enabled
Local setting: No local workspace configuration file.
Effective status: enabled
? Would you like to add Angular routing? No
? Which stylesheet format would you like to use? CSS
            Standalone application structure is new and not yet supported by many existing 'ng add' and 'ng update' integrations with community libraries.
CREATE ilovejavascript.net/README.md (1072 bytes)
CREATE ilovejavascript.net/.editorconfig (274 bytes)
CREATE ilovejavascript.net/.gitignore (548 bytes)
CREATE ilovejavascript.net/angular.json (3015 bytes)
CREATE ilovejavascript.net/package.json (1050 bytes)
CREATE ilovejavascript.net/tsconfig.json (901 bytes)
CREATE ilovejavascript.net/tsconfig.app.json (263 bytes)
CREATE ilovejavascript.net/tsconfig.spec.json (273 bytes)
CREATE ilovejavascript.net/.vscode/extensions.json (130 bytes)
CREATE ilovejavascript.net/.vscode/launch.json (470 bytes)
CREATE ilovejavascript.net/.vscode/tasks.json (938 bytes)
CREATE ilovejavascript.net/src/main.ts (250 bytes)
CREATE ilovejavascript.net/src/favicon.ico (1642 bytes)
CREATE ilovejavascript.net/src/index.html (304 bytes)
CREATE ilovejavascript.net/src/styles.css (80 bytes)
CREATE ilovejavascript.net/src/app/app.component.css (0 bytes)
CREATE ilovejavascript.net/src/app/app.component.html (23083 bytes)
CREATE ilovejavascript.net/src/app/app.component.spec.ts (927 bytes)
CREATE ilovejavascript.net/src/app/app.component.ts (318 bytes)
CREATE ilovejavascript.net/src/app/app.config.ts (117 bytes)
CREATE ilovejavascript.net/src/assets/.gitkeep (0 bytes)
✔ Packages installed successfully.
    Successfully initialized git.
╭ Configuring your application for Cloudflare Step 2 of 3
│ 
├ Installing Angular SSR 
│ installed
│ 
├ Adding Cloudflare Pages adaptor code 
│ added Cloudflare Pages adaptor code`
│ 
├ Installing adaptor dependencies 
│ Installed
│ 
├ Updating application code 
│ Done updating application code
│ 
├ Updating angular.json config 
│ Updated angular.json config
│ 
├ Adding command scripts for development and deployment
│ added commands to `package.json`
│ 
╰ Application configured 

╭ Deploy with Cloudflare Step 3 of 3
│ 
├ Do you want to deploy your application?
│ yes deploying via `npm run deploy`
│
 ⛅️ wrangler 0.0.0-d8f8f3d2 
----------------------------
Getting User settings...

✘ [ERROR] Not logged in.
</details>
</summary>

<summary>
C3 + Qwik
<details>

      ............
    .::: :--------:.
   .::::  .:-------:.
  .:::::.   .:-------.
  ::::::.     .:------.
 ::::::.        :-----:
 ::::::.       .:-----.
  :::::::.     .-----.
   ::::::::..   ---:.
    .:::::::::. :-:.
     ..::::::::::::
             ...::::
     

┌  🦋  Add Integration  cloudflare-pages
│
◇  👻  Ready?  Add cloudflare-pages to your app?
│
│  🐬 Modify
│     - package.json
│     - README.md
│     - .gitignore
│
│  🌟 Create
│     - .node-version
│     - public/_headers
│     - public/_redirects
│     - src/entry.cloudflare-pages.tsx
│     - adapters/cloudflare-pages/vite.config.ts
│
│  💾 Install npm dependency:
│     - wrangler ^2.20.0
│
│  📜 New npm script:
│     - npm run build.server
│     - npm run deploy
│     - npm run serve
│
◇  Ready to apply the cloudflare-pages updates to your app?
│  Yes looks good, finish update!
│
◇  App updated
│
○  New scripts added ─────────╮
│                             │
│     - npm run build.server  │
│     - npm run deploy        │
│     - npm run serve         │
│                             │
├─────────────────────────────╯
│
○  🟣  Next Steps  ─────────────────────────────────────────────────╮
│                                                                             │
│      Now you can build and deploy to Cloudflare Pages with:                 │
│                                                                             │
│       - npm run build: production build for Cloudflare                      │
│       - npm run deploy: it will use the Cloudflare CLI to deploy your site  │
│                                                                             │
├─────────────────────────────────────────────────────────────────────────────╯
│
└  🦄  Success!  Added cloudflare-pages to your app

├ Adding command scripts for development and deployment
│ added commands to `package.json`
│ 
╰ Application configured 

╭ Deploy with Cloudflare Step 3 of 3
│ 
├ Do you want to deploy your application?
│ yes deploying via `npm run pages:deploy`
│
 ⛅️ wrangler 2.20.0 
--------------------
Getting User settings...

✘ [ERROR] Not logged in.
</details>
</summary>

I suspected that my token might be super stale, so I ran:
```
npx wrangler logout
```

After this, I was able to rerun `npm create cloudflare@next`, scaffold an app, and when trying to deploy I was asked to log in and confirm oauth token generation, which ultimately resolved the problem.

This just bit me again. Once again the scenario was that my machine had wrangler logged in on it previously at some point, but not recently, and if I tried to run C3 it failed because we saw the token and assume we are logged in but when the token was used, EWC rejected the request and wrangler exited with:

```
✘ [ERROR] Not logged in.
```

I run `npm create cloudflare` and after all the steps, got an error 
`✘ [ERROR] Not logged in.`

And it doesn't say how to log in. I didn't know that it was a wrangler error. All I knew was that I used C3. How the user is supposed to get out of this situation?

--- a/packages/create-cloudflare/src/helpers/command.ts
+++ b/packages/create-cloudflare/src/helpers/command.ts
@@ -106,6 +106,10 @@ export const runCommand = async (
 						}
 					}
 				});
+
+				cmd.on("error", (code) => {
+					reject(code);
+				});
 			});
 		},
 	});
@@ -255,11 +259,14 @@ export const installWrangler = async () => {
 
 export const isLoggedIn = async () => {
 	const { npx } = detectPackageManager();
-	const output = await runCommand(`${npx} wrangler whoami`, {
-		silent: true,
-	});
-
-	return !/not authenticated/.test(output);
+	try {
+		const output = await runCommand(`${npx} wrangler whoami`, {
+			silent: true,
+		});
+		return /You are logged in/.test(output);
+	} catch (error) {
+		return false;
+	}
 };
 
 export const wranglerLogin = async () => {

